src/content_delegate.js

```diff
@@ -585,31 +588,193 @@ ContentDelegate.prototype.attachRecapLinkToEligibleDocs = function() {
+        PACER.COURT_ABBREVS[this.court],
+        docket_number,
+        document_number,
+      ].join('_').concat('.zip');
+    }
+  };
+
+  // Make the Back button redisplay the previous page.
+  window.onpopstate = function (event) {
+    if (event.state.content) {
+      document.documentElement.innerHTML = event.state.content;
+    }
+  };
+  history.replaceState({content: document.documentElement.innerHTML}, '');
+
+  try {
+    // runtime start
+    $("body").css("cursor", "wait");
+    // fetch the html page which contains the <iframe> link to the zip document.
+
+    // in Firefox, use content.fetch for content-specific fetch requests
+    // https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_scripts#XHR_and_Fetch
+    const browserSpecificFetch = (navigator.userAgent.indexOf('Chrome') < 0) ? content.fetch : window.fetch
+
+    const htmlPage = await browserSpecificFetch(event.data.id).then(res => res.text());
+    console.log("RECAP: Successfully submitted zip file request");
+    const zipUrl = extractUrl(htmlPage);
+
+    //download zip file and save it to chrome storage
+    const nonce = "blob_upload_storage";
+    const zipFile = await fetch(zipUrl)
+      .then(res => res.arrayBuffer())
+      .then(buffer => {
+        console.info('RECAP: Downloaded zip file');
+        saveItemToStorage({[nonce]: arrayBufferToArray(buffer)});
+        return new Blob([buffer], {type: 'application/zip'});
+      })
+    const blobUrl = URL.createObjectURL(zipFile);
+    const pacerCaseId = (event.data.id).match(/caseid\=\d*/)[0].replace(/caseid\=/, "");
+
+    // load options
+    const payload = await getItemsFromStorage(['options']);
+    const options = payload['options'];
+
+    if (options['recap_enabled'] && !this.restricted) {
+      this.recap.uploadZipFile(
+        this.court, // string
+        pacerCaseId, // string
+        nonce, // string
+        (ok) => { // callback
+          if (ok) {
+            const filename = generateFileName(options, pacerCaseId);
+            // convert htmlPage to document
+            const link =
+              `<a id="recap-download" href=${blobUrl} download=${filename} width="0" height="0"/>`;
+            const htmlBody = stringToDocBody(htmlPage);
+            const frame = htmlBody.querySelector('iframe');
+            frame.insertAdjacentHTML('beforebegin', link);
+            frame.src = "";
+            frame.onload = () => document.getElementById('recap-download').click();
+            document.body = htmlBody;
+            history.pushState({content: document.body.innerHTML}, '');
+            // show notifier
+            this.notifier.showUpload('Zip uploaded to the Public Recap Archive', () => {});
+            saveItemToStorage({[nonce]: ""});
+            saveItemToStorage({'docId': ""});
+          }
+        }
+      );
     }
-  }, this));
+  } catch (err) {
+    console.error(err);
+  }
 };
+
+// Same as handlePDFView, but for zip files!
```

Comment:
I **think** you mean this:

```suggestion
// Same as handleSingleDocumentPageView, but for zip files
```